rust/kernel/chrdev.rs

```diff
@@ -20,10 +20,66 @@ use crate::error::{Error, KernelResult};
 use crate::file_operations;
 use crate::types::CStr;
 
+struct SafeCdev(*mut bindings::cdev);
```

Comment:
nit: I don't think we need this `Safe` prefix. How about just  `Cdev` or `CharDev`?